validation.lua

```diff
@@ -308,17 +317,18 @@ assert_collection_exists = function (self)
 end
 
 assert_can_view_collection = function (self, collection)
```

Comment:
I found the old logic a little confusing to parse... 

I also wonder if it makes sense to write `collection:user_can_view()` and call that in `assert_can_view_collection`